pkg/controller/bootimage/boot_image_controller_test.go

```
@@ -1083,6 +1084,103 @@ func TestResetClusterBootImage(t *testing.T) {
 	}
 }
 
+// TestUpdateConditionsDegradedCrossResource verifies that a Degraded condition set by one resource
+// sync is not cleared by a subsequent clean sync from another resource, and that all per-resource
+// sync errors appear in the condition message regardless of which resource calls last.
+func TestUpdateConditionsDegradedCrossResource(t *testing.T) {
+	cases := []struct {
+		name               string
```

golangci-lint (2.12.2)

[error] File is not properly formatted

(gofmt)